lib/devcycle-ruby-server-sdk/api/dev_cycle_provider.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module DevCycle
+  class Provider
+    attr_reader :client
+    def initialize(client)
+      unless client.is_a?(DevCycle::Client)
+        fail ArgumentError('Client must be an instance of DevCycleClient')
+      end
+      @client = client
+    end
+
+    def init
+      # We handle all initialization on the DVC Client itself
+    end
+
+    def shutdown
+      @client.close
+    end
+
+    def fetch_boolean_value(flag_key:, default_value:, evaluation_context: nil)
+      # Retrieve a boolean value from provider source
+      @client.variable(Provider.user_from_openfeature_context(evaluation_context), flag_key, default_value)
```

Comment:
don't these need to be `variableValue`?

Reply:
No - the value is directly exposed

Reply:
Are we doing a similar casting like this somewhere? 
<img width="564" alt="image" src="https://github.com/user-attachments/assets/1d86f6b1-9baa-4260-a0bc-0867ee3c8d3d" />

Reply:
The resolutions don't exist yet in the ruby sdk